pyHS100/pyHS100.py

```diff
@@ -318,12 +318,20 @@ def query(host, request, port=9999):
 
         sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         sock.connect((host, port))
+        _LOGGER.debug("> (%i) %s" % (len(request), request))
         sock.send(TPLinkSmartHomeProtocol.encrypt(request))
-        buffer = sock.recv(4096)[4:]
+        buffer = bytes()
+        while True:
+            chunk = sock.recv(4096)
+            buffer += chunk
+            #_LOGGER.debug("Got chunk %s" % len(chunk))
+            if len(chunk) == 0:
```

Comment:
Good catch, didn't recall for sure whether empty b'' would've matched that, but it does.